mlpipeline/encoder_service/app.py

```
@@ -0,0 +1,98 @@
+
+
+app = FastAPI(title="User Encoder Service")
+
+_MODEL: Optional[UserEncoder] = None
+_INPUT_DIM: Optional[int] = None
+
+
+def _load_model(ckpt_path: str, input_dim: int, device: torch.device) -> UserEncoder:
+    cfg = UserEncoderConfig()
+    if input_dim != cfg.user_input_dim:
+        raise ValueError(f"the pretrained in_dim is {cfg.user_input_dim}, but we get {input_dim}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)